internal/container/default_registry.go

```diff
@@ -0,0 +1,50 @@
+package container
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/docker/distribution/reference"
+)
+
+// Produces a new image name that is in the specified registry.
+// The name might be ugly, favoring uniqueness and simplicity and assuming that the prettiness of ephemeral dev image
+// names is not that important.
+func replaceNamedTagged(defaultRegistry string, name reference.NamedTagged) (reference.NamedTagged, error) {
+	newN, err := replaceNamed(defaultRegistry, MustParseNamed(name.Name()))
```

Comment:
instead of serializing and de-serializing and panicking on error, you'd usually use `reference.TrimNamed(name)` here